### DISCUSSION TITLE
JSON-Scada on RaspberryPI

### DISCUSSION THREAD

**Discussion by barsicepy** [riclolsen/json-scada#42] [Q&A] [answered]
Tell me please. I want use JSON-SCADA as protocol gateway with minimal scaling calculations.
How can I install JSON-SCADA on RaspbianOS (ARM32 Debian 9), if [MongoDB Server (4.2 and later)](https://docs.mongodb.com/manual/installation/#upgrade-community-edition-to-enterprise-edition-tutorials) and [PostgreSQL 12](https://www.postgresql.org/download/linux/debian/) supports only ARM64.

Is it possible to use JSON-SCADA on RaspbianOS? Or I need use Ubuntu Core or something else? What OS do you recommend for JSON-SCADA with RaspberryPI.

**Comment by riclolsen** [comment, marked as answer]
Hi!
Which protocols do you intend to use?
DNP3 is currently Windows-only.

**Reply by riclolsen** [reply]
I think you need 64 bit Ubuntu for MongoDB. Postgresql you may not need for just gateway function.

    https://www.mongodb.com/developer/how-to/mongodb-on-raspberry-pi/

There is no automatic installation process for Linux.

You need to install all the requirements for your platform.

Then clone the repo and build.

Next, use Supervisor to configure the processes you need. See example and instructions here

    https://github.com/riclolsen/json-scada/blob/master/docs/install.md#manual-installation

**Comment by barsicepy** [comment]
Right now, I need MQTT и Modbus. In future - IEC60870-101/104.
Ok, I see. But Raspberry 3 has 1Gb RAM, it's not enough for Ubuntu 64bit. It's seems RaspberryPI 4 with 4Gb RAM is minimal requirement.
Thank you for your help.

**Reply by riclolsen** [reply]
Modbus is currently monitoring mode only, must use Telegraf to scan.
Also one should consider that JSON-SCADA persists any incoming data to MongoDB, this can mean a lot of writes to the disk. The SD card has a limited write capacity, so an SSD may be necessary at least for the MongoDB data folder.

**Reply by SonGoKu-123** [reply]
> Modbus is currently monitoring mode only, must use Telegraf to scan.
> Also one should consider that JSON-SCADA persists any incoming data to MongoDB, this can mean a lot of writes to the disk. The SD card has a limited write capacity, so an SSD may be necessary at least for the MongoDB data folder.

In the near future, Is it possible to use JSON-SCADA as a gateway between modbus-IEC104 protocols on RaspbianOS?

**Reply by riclolsen** [reply]
MongoDB can only possibly work on 64bit OSs, I am not sure RaspbianOS 64bits is compatible.
On Ubuntu 64 it is possible now with Modbus read-only (no writing).
Eventually, a native Modbus driver will allow for commands also.

**Reply by SonGoKu-123** [reply]
Thank you @riclolsen , 
Hope to see native Modbus driver soon.
